passkey-authenticator/src/user_validation.rs

```diff
@@ -46,8 +66,42 @@ impl MockUserValidationMethod {
             .returning(|| Some(true))
             .times(times);
         user_mock
-            .expect_check_user_verification()
-            .returning(|| Box::pin(async { true }))
+            .expect_check_user()
+            .with(
+                mockall::predicate::always(),
+                mockall::predicate::eq(true),
+                mockall::predicate::eq(true),
+            )
+            .returning(|_, _, _| {
+                Ok(UserCheck {
+                    presence: true,
+                    verification: true,
+                })
+            })
+            .times(times);
+        user_mock
+    }
+
+    /// Sets up the mock for returning true for the verification.
+    pub fn verified_user_with_credential(times: usize, credential: Passkey) -> Self {
+        let mut user_mock = MockUserValidationMethod::new();
+        user_mock
+            .expect_is_verification_enabled()
+            .returning(|| Some(true))
+            .times(times);
+        user_mock
+            .expect_check_user()
+            .with(
+                mockall::predicate::eq(Some(credential)),
+                mockall::predicate::eq(true),
+                mockall::predicate::eq(true),
+            )
```

Comment:
**Suggestion(2 of 4):** Mockall also does not work very well here and the `with` clause needs to change to using `withf`:

```suggestion
            .withf(move |cred, up, uv| cred == &Some(&credential) && *up && *uv)
```

Reply:
This was not used so I removed it